### PR TITLE
improve dm document

### PR DESCRIPTION
**Pull request by IANTHEREAL**
### What problem does this PR solve? 
refine dm document
we should clarify some main concepts in overview
like units and privileges

### Check List 

Tests 

 - No code

### CODE REVIEW

File: docs/dm/zh_CN/privileges.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
+DM 使用权限说明
+===
+
+### DM-Worker 需要的权限
+包含 relay log，dump，load，replicate binlog 等 units， 这里先总体说下 上下游分别需要什么权限；
```
Comment (shenli) [inline docs/dm/zh_CN/privileges.md]: units 是什么？

File: docs/dm/zh_CN/privileges.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+INSERT
+UPDATE
+DELETE
+CREATE
+DROP
+ALTER
+INDEX
+即：GRANT SELECT,INSERT,UPDATE,DELETE,CREATE,DROP,ALTER,INDEX  ON *.* TO 'your_user'@'your_wildcard_of_host';
+
+因为 DM 包含 dump，load，replicate binlog events 等组件，下面会对各组件进行权限的细分，方便大家深入理解（注意，随着需求的变化，这些权限也会跟着变化，并非一成不变）
```
Comment (shenli) [inline docs/dm/zh_CN/privileges.md]: “方便大家深入理解” 去掉，这个偏口语

**Comment by IANTHEREAL** [comment]
@shenli PTAL again

**Comment by IANTHEREAL** [comment]
@kennytm PTAL

File: docs/dm/zh_CN/change-log.md [hunk 3]
```diff
@@ -5,7 +5,7 @@ Change Log
 
 #### bug 修复
 
-- 修复下游 PD, TiKV 等 timeout 时 syncer 未 retry 的问题
+- 修复下游 PD, TiKV 等 timeout 时 binlog replication unit 未 retry 的问题
```
Comment (kennytm) [inline docs/dm/zh_CN/change-log.md]: There's no Chinese term for "binlog replication unit"?

File: docs/dm/zh_CN/dm-worker-unit.md [hunk 4]
```diff
@@ -0,0 +1,120 @@
+DM-Worker 详细介绍
+===
+
+### 简介
+- DM-Worker 可以链接到一台 MySQL/MariDB，并且注册为该实例的 Slave
```
Comment (kennytm) [inline docs/dm/zh_CN/dm-worker-unit.md]:
```suggestion
- DM-Worker 可以链接到一台 MySQL/MariaDB，并且注册为该实例的 Slave
```

File: docs/dm/zh_CN/dm-worker-unit.md [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+- 支持读取 MySQL/MariaDB Binlog 持久化保存再本地（relay log）
```
Comment (kennytm) [inline docs/dm/zh_CN/dm-worker-unit.md]:
```suggestion
- 支持读取 MySQL/MariaDB Binlog 持久化保存在本地（relay log）
```

File: docs/dm/zh_CN/dm-worker-unit.md [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+#### loader
+读取 dump unit 的数据文件，然后加载到下游 TiDB
+
+#### binlog replication/syncer
+读取 relay log unit 的 Binlog events，转化为 SQLs，然后应用到下游 TiDB
+
+
+
+### DM-Worker 需要的权限
+包含 relay log，dump，load，replicate binlog 等任务运行单元， 这里先总体说下 上下游分别需要什么权限；
```
Comment (kennytm) [inline docs/dm/zh_CN/dm-worker-unit.md]:
```suggestion
包含 relay log、dump、load、replicate binlog 等任务运行单元， 这里先总体说下 上下游分别需要什么权限；
```

File: docs/dm/zh_CN/dm-worker-unit.md [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+
+#### 上游（mysql/mariadb）
```
Comment (kennytm) [inline docs/dm/zh_CN/dm-worker-unit.md]:
```suggestion
#### 上游（MySQL/MariaDB）
```

File: docs/dm/zh_CN/dm-worker-unit.md [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+SELECT
+RELOAD
+RELICATION SLAVE
+REPLICATION CLIENT
+
+即 GRANT SELECT,RELOAD,REPLICATION SLAVE, REPLICATION CLIENT  ON *.* TO 'your_user'@'your_wildcard_of_host';
```
Comment (kennytm) [inline docs/dm/zh_CN/dm-worker-unit.md]:
``````suggestion
即 

```sql
GRANT SELECT,RELOAD,REPLICATION SLAVE, REPLICATION CLIENT  ON *.* TO 'your_user'@'your_wildcard_of_host';
```
``````

File: docs/dm/zh_CN/dm-worker-unit.md [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+
+#### 下游 (tidb)
```
Comment (kennytm) [inline docs/dm/zh_CN/dm-worker-unit.md]:
```suggestion
#### 下游 (TiDB)
```

File: docs/dm/zh_CN/dm-worker-unit.md [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+INSERT
+UPDATE
+DELETE
+CREATE
+DROP
+ALTER
+INDEX
+即：GRANT SELECT,INSERT,UPDATE,DELETE,CREATE,DROP,ALTER,INDEX  ON *.* TO 'your_user'@'your_wildcard_of_host';
+
+因为 DM 包含 dump，load，replicate binlog events 等组件，下面会对各组件进行权限的细分（注意，随着需求的变化，这些权限也会跟着变化，并非一成不变）
```
Comment (kennytm) [inline docs/dm/zh_CN/dm-worker-unit.md]:
```suggestion
因为 DM 包含 dump、load、replicate binlog events 等组件，下面会对各组件进行权限的细分（注意，随着需求的变化，这些权限也会跟着变化，并非一成不变）
```

File: docs/dm/zh_CN/dm-worker-unit.md [hunk 11]
```diff
@@ -0,0 +1,120 @@
...
+#### 下游 (tidb)
+SELECT 
+INSERT
+UPDATE
+DELETE
+CREATE
+DROP
+ALTER
+INDEX
+即：GRANT SELECT,INSERT,UPDATE,DELETE,CREATE,DROP,ALTER,INDEX  ON *.* TO 'your_user'@'your_wildcard_of_host';
```
Comment (kennytm) [inline docs/dm/zh_CN/dm-worker-unit.md]:
``````suggestion
即：

```sql
GRANT SELECT,INSERT,UPDATE,DELETE,CREATE,DROP,ALTER,INDEX  ON *.* TO 'your_user'@'your_wildcard_of_host';
```
``````

File: docs/dm/zh_CN/dm-worker-unit.md [hunk 12]
```diff
@@ -0,0 +1,120 @@
...
+
+因为 DM 包含 dump，load，replicate binlog events 等组件，下面会对各组件进行权限的细分（注意，随着需求的变化，这些权限也会跟着变化，并非一成不变）
+
+
+
+### units 需要的最小权限
+
+#### relay log
+
+###### 上游（mysql/mariadb）
```
Comment (kennytm) [inline docs/dm/zh_CN/dm-worker-unit.md]:
```suggestion
##### 上游（MySQL/MariaDB）
```

File: docs/dm/zh_CN/dm-worker-unit.md [hunk 13]
```diff
@@ -0,0 +1,120 @@
...
+SELECT （查询上游的一些环境变量，比如 binlog_format）
+REPLICATION SLAVE (拉取 binlog)
+REPLICATION CLIENT (show master status, show slave status)
+
+###### 下游 (tidb)
```
Comment (kennytm) [inline docs/dm/zh_CN/dm-worker-unit.md]: ditto

File: docs/dm/zh_CN/dm-worker-unit.md (hunk 12 above)
Comment (kennytm) [reply docs/dm/zh_CN/dm-worker-unit.md]: And seriously do you need a 5th-level heading? I guess a table is more appropriate.

| 組件 | 上游 (MySQL/MariaDB) | 下游 (TiDB) | 系統 |
|----:|:--------------------|:------------|:----|
| relay log | SELECT ...<br>REPLICATION SLAVE ...<br>REPLICATION CLIENT ... | 无 | 本地读写磁盘权限 |

File: docs/dm/zh_CN/overview.md [hunk 14]
```diff
@@ -0,0 +1,42 @@
+DM 简介
+===
+
+### 介绍
+
+DM (Data Migration) 是一体化数据同步任务管理平台，支持全量备份和 Mariadb/MySQL binlog 增量同步，设计的主要目的是
```
Comment (kennytm) [inline docs/dm/zh_CN/overview.md]:
```suggestion
DM (Data Migration) 是一体化数据同步任务管理平台，支持全量备份和 MariaDB/MySQL binlog 增量同步，设计的主要目的是
```

File: docs/dm/zh_CN/overview.md [hunk 15]
```diff
@@ -0,0 +1,42 @@
...
+   - 标准化 （e.g. 工具运行，错误定义）
```
Comment (kennytm) [inline docs/dm/zh_CN/overview.md]:
```suggestion
   - 标准化 （例如 工具运行，错误定义）
```

File: docs/dm/zh_CN/dm-worker-unit.md (hunk 12 above)
Comment (shenli) [reply docs/dm/zh_CN/dm-worker-unit.md]: Agree

File: docs/dm/zh_CN/change-log.md (hunk 3 above)
Comment (IANTHEREAL) [reply docs/dm/zh_CN/change-log.md]: `binlog replication` is name of a unit, we should keep binlog replication, but we can rename it to `binlog replication 处理单元`

File: docs/dm/zh_CN/dm-worker-unit.md (hunk 12 above)
Comment (IANTHEREAL) [reply docs/dm/zh_CN/dm-worker-unit.md]: nice Kenny, I always wanted to do it

File: docs/dm/zh_CN/dm-worker-unit.md [hunk 16]
```diff
@@ -0,0 +1,67 @@
...
+即：
+
+```sql
+GRANT SELECT,INSERT,UPDATE,DELETE,CREATE,DROP,ALTER,INDEX  ON *.* TO 'your_user'@'your_wildcard_of_host';
+```
+
+下面会对各处理单元进行权限的细分（注意，随着需求的变化，这些权限也会跟着变化，并非一成不变）
+
+
+### units 需要的最小权限
```
Comment (kennytm) [inline docs/dm/zh_CN/dm-worker-unit.md]: units → 处理单元?

**Review by kennytm — APPROVED** [review]
Rest LGTM

File: docs/dm/zh_CN/dm-worker-unit.md (hunk 16 above)
Comment (shenli) [reply docs/dm/zh_CN/dm-worker-unit.md]: 我们用 component/组件 来替换 unit/单元 如何？

Comment (IANTHEREAL) [reply docs/dm/zh_CN/dm-worker-unit.md]: `组件/component` 已经用来表示 dm-master， dm-worker。 有没有比处理单元更好的名字，这些 unit 的含义是任务的逻辑处理环节单元。

**Comment by IANTHEREAL** [comment]
@kennytm @csuzhangxc  PTAL

**Review by kennytm — COMMENTED** [review]
lgtm

File: docs/dm/zh_CN/dm-worker-unit.md [hunk 17]
```diff
@@ -0,0 +1,69 @@
+DM-Worker 详细介绍
```
Comment (csuzhangxc) [inline docs/dm/zh_CN/dm-worker-unit.md]: should we need unify case (upper / lower) for component / unit names?

File: docs/dm/zh_CN/dm-worker-unit.md [hunk 18]
```diff
@@ -0,0 +1,69 @@
...
+===
+
+### 简介
+- DM-Worker 可以链接到一台 MySQL/MariaDB，并且注册为该实例的 Slave
```
Comment (csuzhangxc) [inline docs/dm/zh_CN/dm-worker-unit.md]: `链接` or `连接`?

File: docs/dm/zh_CN/dm-worker-unit.md [hunk 19]
```diff
@@ -0,0 +1,69 @@
...
+
+
+### DM-Worker 需要的权限
+包含 relay log、dump、load、replicate binlog 等处理单元， 这里先总体说下 上下游分别需要什么权限；
+
+#### 上游（MySQL/mariaDB)
+SELECT
+RELOAD
+RELICATION SLAVE
+REPLICATION CLIENT
```
Comment (csuzhangxc) [inline docs/dm/zh_CN/dm-worker-unit.md]: new white spaces or `-` needed to separate these lines.
you can view the rendered html to see the problem

File: docs/dm/zh_CN/dm-worker-unit.md [hunk 20]
```diff
@@ -0,0 +1,69 @@
...
+
+#### 下游 (TiDB)
+SELECT 
+INSERT
+UPDATE
+DELETE
+CREATE
+DROP
+ALTER
+INDEX
```
Comment (csuzhangxc) [inline docs/dm/zh_CN/dm-worker-unit.md]: ditto

File: docs/dm/zh_CN/maintenance/metrics-alert.md [hunk 21]
```diff
@@ -5,247 +5,74 @@ DM 监控与告警
...
-
- 
+| metric 名称 | 说明 | 告警说明 |
+|----:|:------------|:----|
+| task state | 同步子任务的状态 | 当子任务状态处于 paused 超过 10 分钟时|
+
+
+### relay log
+
+| metric 名称 | 说明 | 报警说明 |
```
Comment (csuzhangxc) [inline docs/dm/zh_CN/maintenance/metrics-alert.md]: `报警` or `告警`?
another two below

File: docs/dm/zh_CN/dm-worker-unit.md (hunk 17 above)
Comment (IANTHEREAL) [reply docs/dm/zh_CN/dm-worker-unit.md]: still not do it

**Review by csuzhangxc — APPROVED** [review]
LGTM

**Pull request merged**